powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/callbacks/CallbackUrlRestClientCacheLoader.java

```diff
@@ -22,108 +41,65 @@
 import org.springframework.security.oauth2.core.AuthorizationGrantType;
 import org.springframework.stereotype.Component;
 
-import java.util.Map;
-import java.util.concurrent.ConcurrentHashMap;
+import java.time.LocalDateTime;
+import java.util.Optional;
 
 /**
- * Manager class for REST Client instances used to Callback URL Event processing.
+ * Specialization of {@link CacheLoader} for {@link CachedRestClient}.
  *
  * @author Jan Pesek, jan.pesek@wultra.com
  */
 @AllArgsConstructor
-@Component
 @Slf4j
-public class CallbackUrlRestClientManager {
+@Component
+public class CallbackUrlRestClientCacheLoader implements CacheLoader<String, CachedRestClient> {
 
     private final PowerAuthServiceConfiguration powerAuthServiceConfiguration;
     private final CallbackUrlAuthenticationEncryptor callbackUrlAuthenticationEncryptor;
-    private final CallbackUrlEventRepository callbackUrlEventRepository;
+    private final CallbackUrlRepository callbackUrlRepository;
 
-    // Store REST clients in cache with their callback ID as a key
-    private final Map<String, RestClient> restClientCache = new ConcurrentHashMap<>();
-    private final Object restClientCacheLock = new Object();
+    @Override
+    public @Nullable CachedRestClient load(final String callbackUrlId) throws RestClientException, GenericServiceException {
+        logger.debug("Loading RestClient for CallbackUrl: id={}", callbackUrlId);
 
-    /**
-     * Get a REST Client using which a Callback URL Event will be dispatched.
-     * @param callbackUrlEvent Callback URL Event for which to get the REST Client.
-     * @return REST Client.
-     * @throws RestClientException In case the REST Client initialization fails.
-     */
-    public RestClient getRestClient(final CallbackUrlEvent callbackUrlEvent) throws RestClientException, GenericServiceException {
-        final String cacheKey = callbackUrlEvent.restClientCacheKey();
-        synchronized (restClientCacheLock) {
-            final RestClient restClient = restClientCache.get(cacheKey);
-            if (restClient == null) {
-                logger.debug("REST client not found in cache, initializing new REST client, callback cache key: {}", cacheKey);
-                return createRestClientAndStoreInCache(callbackUrlEvent);
-            } else {
-                logger.debug("REST client found in cache, callback cache key: {}", cacheKey);
-                return restClient;
-            }
+        final Optional<CallbackUrlEntity> optionalCallbackUrlEntity = callbackUrlRepository.findById(callbackUrlId);
+        if (optionalCallbackUrlEntity.isEmpty()) {
+            logger.debug("CallbackUrlEntity is not available: id={}", callbackUrlId);
```

Comment:
In the push-server, we have changed this kind of log to WARN level.
cc @zcgandcomp

Reply:
Yes, I suggest to change to warn.